File: DICCIONARIOS/Java/4.java
import java.util.*;

/**
 * Ejercicio 4 — Conversión de fecha (Java)
 * Convierte 'dd/mm/aaaa' a 'dd de <mes> de aaaa' usando un arreglo de meses.
 */
public class FormatoFecha {
    public static void main(String[] args) {
        String[] meses = {"", "enero","febrero","marzo","abril","mayo","junio",
                          "julio","agosto","septiembre","octubre","noviembre","diciembre"};
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese fecha (dd/mm/aaaa): ");
        String fecha = sc.nextLine().trim();
        try {
            String[] p = fecha.split("/");
            int d = Integer.parseInt(p[0]);
            int m = Integer.parseInt(p[1]);
            String a = p[2];
            if (m >= 1 && m <= 12) {
                System.out.println(d + " de " + meses[m] + " de " + a);
            } else {
                System.out.println("Mes inválido");
            }
        } catch (Exception e) {
            System.out.println("Formato inválido. Ejemplo: 24/08/2025");
        }
    }
}

File: DICCIONARIOS/Python/4.py
#!/usr/bin/env python3
# Ejercicio 4 — Conversión de fecha (Python)
# Entrada: 'dd/mm/aaaa'. Salida: 'dd de <mes> de aaaa'.
# Uso de diccionario para mapear número de mes a su nombre.

MESES = {
    1: "enero", 2: "febrero", 3: "marzo", 4: "abril",
    5: "mayo", 6: "junio", 7: "julio", 8: "agosto",
    9: "septiembre", 10: "octubre", 11: "noviembre", 12: "diciembre"
}

def formatear_fecha(fecha: str) -> str:
    """Convierte 'dd/mm/aaaa' en 'dd de <mes> de aaaa'. Valida el mes."""
    d, m, a = fecha.split("/")
    dia, mes = int(d), int(m)
    if 1 <= mes <= 12:
        return f"{dia} de {MESES[mes]} de {a}"
    return "Mes inválido"

if __name__ == "__main__":
    f = input("Ingrese fecha (dd/mm/aaaa): ").strip()
    try:
        print(formatear_fecha(f))
    except Exception:
        print("Formato inválido. Ejemplo correcto: 24/08/2025")
